core/payment-driver/zksync/examples/withdrawal.rs

```diff
@@ -0,0 +1,111 @@
+#[macro_use]
+extern crate log;
+
+use bigdecimal::BigDecimal;
+use hex::ToHex;
+use structopt::StructOpt;
+use ya_zksync_driver::zksync::{faucet, utils};
+use zksync::zksync_types::{TxFeeTypes, H256};
+use zksync::{types::BlockStatus, Network, Provider, Wallet, WalletCredentials};
+use zksync_eth_signer::{EthereumSigner, PrivateKeySigner};
+
+const TOKEN: &str = "GNT";
+const PRIVATE_KEY: &str = "312776bb901c426cb62238db9015c100948534dea42f9fa1591eff4beb35cc13";
+
+#[derive(Clone, Debug, StructOpt)]
+struct Args {
+    #[structopt(long = "gen-key")]
+    genkey: bool,
+
+    #[structopt(long, default_value = "5.0")]
+    amount: f64,
+}
+
+#[actix_rt::main]
+async fn main() -> anyhow::Result<()> {
+    let log_level = std::env::var("RUST_LOG").unwrap_or("info".to_owned());
+    std::env::set_var("RUST_LOG", log_level);
+    env_logger::init();
+
+    let args: Args = Args::from_args();
+    let private_key = if args.genkey {
+        debug!("Using randomly generated key");
+        H256::random()
+    } else {
+        debug!("Using hardcoded key");
+        let mut bytes = [0u8; 32];
+        hex::decode_to_slice(PRIVATE_KEY, &mut bytes)
+            .expect("Cannot decode bytes from hex-encoded PK");
+        H256::from(bytes)
+    };
+
+    let pk_hex: String = private_key.encode_hex();
+    let signer = PrivateKeySigner::new(private_key);
+    let address = signer.get_address().await?;
+    let addr_hex = format!("0x{}", address.encode_hex::<String>());
+    info!("Private key: {}\nAccount address {}", pk_hex, addr_hex);
```

Comment:
I would try to avoid `\n` in log lines, unfortunately its used more often in yagna, but logs are cleaner without it.

We have logging guidelines that propose a log message structure like this:
```
This is what happened. key1={value1}, key2={value2}
```

Source: https://github.com/golemfactory/golem-internal/wiki/Logging-guidelines#log-message-structure

Because this is just an example it does not matter that much, just FYI :)

Reply:
Note to self: TODO: Copy link to guidelines